addons/stock/models/stock_inventory.py

```diff
@@ -319,6 +322,32 @@ def _get_inventory_lines_values(self):
             vals += self._get_exhausted_inventory_lines_vals({(l['product_id'], l['location_id']) for l in vals})
         return vals
 
+    @api.model
+    def _run_cyclic_inventory_tasks(self, company_id=False):
+        """ Generate and clean up cyclic inventories for locations when:
+            - [to create] The location has a cyclic count set (i.e. inventory every XX days)
+            - [to delete] The location's inventory has not started before current day (i.e. still draft) [new one created afterwards]
+        """
+        domain = [('next_inventory_date', '<=', fields.Date.today())]
+        if company_id:
+            # manually triggered => only apply to user's companies
+            domain = expression.AND([[('company_id', '=', company_id)], domain])
+        else:
+            # cron triggered => apply to all companies
+            domain = expression.AND([[('company_id', '!=', False)], domain])
+        locations = self.env['stock.location'].search(domain)
+        if locations:
+            self.search([('state', '=', 'draft'), ('location_ids', 'in', locations.ids), ('product_ids', '=', False), ('create_date', '<', fields.Datetime.today())]).unlink()
+            locations_w_inv = self.search([('state', 'in', ['draft', 'confirm']), ('location_ids', 'in', locations.ids), ('product_ids', '=', False)]).mapped('location_ids')
+            # Don't create overlapping inventories for a location + its children or new inventories when they've already been made
+            locations = locations.filtered(lambda l: l.location_id not in locations and l not in locations_w_inv)
```

Comment:
It strange to check only for one parent. e.g:
- Inventory for WH/Stock
- Inventory for WH/Stock/Shelf1
- Inventory for WH/Stock/Shelf1/Drawer1.

If we put a cyclic count on WH/Stock/Shelf1/Drawer1 and WH/Stock/
We will have 2 inventories.

But if we add another on the intermedidate WH/Stock/Shelf1.
We will have only one.

Reply:
It would be annoying if we already have a current inventory for:
- Inventory for WH/Stock/Shelf1
Then we could create a new one for Inventory for WH/Stock that would conflict with the existing one.